benchmark/src/main/java/com/linecorp/decaton/benchmark/BenchmarkRmi.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.decaton.benchmark;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.rmi.AlreadyBoundException;
+import java.rmi.NotBoundException;
+import java.rmi.Remote;
+import java.rmi.RemoteException;
+import java.rmi.registry.LocateRegistry;
+import java.rmi.server.RMIServerSocketFactory;
+import java.rmi.server.RMISocketFactory;
+import java.rmi.server.UnicastRemoteObject;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.LinkedBlockingQueue;
+
+import com.linecorp.decaton.benchmark.Execution.Stage;
+
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Remote interface used for communication between forked execution and benchmark controller.
+ *
+ * {@link Stage} changes are notified through {@link BenchmarkRmi#pollStage()} rather than
+ * RMI callback directly which requires benchmark controller to be concurrent aware.
+ */
+@Slf4j
+class BenchmarkRmi implements AutoCloseable {
+    private static final String REMOTE_NAME = "BenchmarkRmi";
+
+    private final BlockingQueue<Stage> stageQueue = new LinkedBlockingQueue<>();
```

Comment:
That's because we have to specify capacity if we use ArrayBlockingQueue.

Yeah, in current implementation, there should be only 3 stage callback invocation at maximum (READY_WARMUP => READY => FINISH), but it sounds natural to use unbounded queue this time rather than specifying "sufficiently enough" capacity?

Reply:
I see. I prefer to put explicit capacity bound for all cases unless I really want it to potentially consume all available heap. so to me, Array with some sufficiently large cap (like 16 or 32) still makes sense, but it's fine to leave it as-is if it's backed by certain intention :)